### PR TITLE
Tec 45 eseway payment gateway integration

### PR DESCRIPTION
**Pull request by RORingBBK**
- Initial eSewa payment gateway setup.

### CODE REVIEW

File: app/models/spree/gateway/esewa.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Spree
+  class Gateway::Esewa < Gateway
+    preference :mode, :string
+    preference :sandbox_url, :string
+    preference :production_url, :string
+    preference :scd, :string, default: 'epay_payment'
```
[the remark this answers is not in the document]
Comment (RORingBBK) [reply app/models/spree/gateway/esewa.rb]: SCD will be the merchant code that will be provided by esewa.
More read: https://developer.esewa.com.np/#/epay?id=payment

File: app/models/spree/gateway/esewa.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    preference :username, :string
```
[the remark this answers is not in the document]
Comment (RORingBBK) [reply app/models/spree/gateway/esewa.rb]: Agreed. We do not need to save the credentials. I have updated accordingly on the recent changes.

File: app/models/spree/gateway/esewa.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module Spree
+  class Gateway::Esewa < Gateway
+    preference :mode, :string
```
[the remark this answers is not in the document]
Comment (RORingBBK) [reply app/models/spree/gateway/esewa.rb]: I have removed the `mode` attribute from here. There is already an existing `test_mode` attribute provided with Spree for PaymentMethods. Should we stick with the `test_mode` option? Thoughts?

[the remark this answers is not in the document]
Comment (poudelprakash) [reply app/models/spree/gateway/esewa.rb]: yeah lets stick with test_mode

File: app/services/spree/payment/epay_service.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      def payment
+        response = connection.post(api_url, request_body)
+        response.body
+      end
+
+      private
+
+      # Setting to true will log requests and response to $stdout,
+      # should always be set to false when commiting code
+      DEBUG_MODE = false
```
Comment (poudelprakash) [inline app/services/spree/payment/epay_service.rb:18]: if this is related to printing errors, lets utilise debug level config(config.log_level) from config/development.rb or production.rb..
and act accordingly
(sorry for not putting down specific syntax implementation)

**Comment by poudelprakash** [comment]
Merged in favor of https://github.com/TruemarkDev/spree_gateways_nepal/pull/7

**Pull request closed without merging**